### ISSUE TITLE
Paper API v2 doesn't support "latest" version

### ISSUE THREAD

**Issue by Cookie04DE** [Phyremaster/papermc-docker#13]
V1 of the API will only be up for another 3 months (as stated by developers on their discord) and v2 doesn't support getting the latest version anymore.

**Comment by Phyremaster** [comment]
That is a bit of a problem. I'll try to take a look into that later this week.

**Comment by Phyremaster** [comment]
The semester is over, so I can fix this now. It looks like [this](https://papermc.io/api/docs/swagger-ui/index.html?configUrl=/api/openapi/swagger-config#/) contains all of the necessary information to rewrite the downloading/version checking code. I'll get on this.

**Comment by Phyremaster** [comment]
...Except it doesn't mention what the name of the file is, so the download GET request is virtually useless. I'll have to hop in their Discord and ask.

**Comment by Lektonic** [comment]
So, I reviewed the API a bit, and it seems that they've made it to where you need to do a bit more parsing to get the information. For example, you can get the latest version as of the time of this writing by going to:

`https://papermc.io/api/v2/projects/paper/versions/1.16.4/builds/332/downloads/paper-1.16.4-332.jar`

(Notice how the file name is just <project>-<version>-<build>.jar)

You can only get this result by building in each download, or parsing the JSON of the upper paths too.

Parsing JSON in pure bash isn't really possible though. You'd have to grab a tool such as [jq](https://stedolan.github.io/jq/download/) before you did anything though.  Luckily that's just a 3MB uncompressed executable file you can download at first run along with paper.

Long story short, based off of the new API, you might need a script that's a bit more heavy and uses a secondary outside tool to parse JSON.

If this is something you would like some help with, let me know.

**Comment by Phyremaster** [comment]
I'll throw a small program together to handle it. It shouldn't be too hard, but it'll probably take me a week or two to get around to it.

**Comment by Phyremaster** [comment]
Thank you very much! In the future, I'd like to try to add a "latest" option for MC version as well, which should mostly work the same
